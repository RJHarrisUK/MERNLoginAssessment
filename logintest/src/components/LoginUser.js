import React, { Component } from 'react';
import axios from 'axios';

export default class LoginUser extends Component {

    constructor() {
        super();
        this.state = {
            data: [],
            error: "",
        };
    }

    loginUser = (event) => {

        event.preventDefault();
        console.log(event.target[0].value)

        let loginDetails = {
            username: event.target[0].value,
            email: event.target[1].value,
            password: event.target[2].value,
        }

        axios
            .post("http://localhost:5000/User/loginUser", loginDetails)
            .then(response => {

                this.setState({ "error": (response.data) })

                console.log(response.data)

            }).catch(err => console.log(err))
    }

    render() {

        return (
            <div>
                <h1>Login User</h1>
                <form onSubmit={this.loginUser}>
                    <input id="username" type="text" placeholder="Username"></input>
                    <br></br>
                    <input id="email" type="text" placeholder="E-mail"></input>
                    <br></br>
                    <input id="password" type="text" placeholder="Password"></input>
                    <p style={{ color: 'red' }}>{this.state.error}</p>
                    <br></br>
                    <button type="submit" >Login User</button>
                </form>
            </div>
        )
    }

}