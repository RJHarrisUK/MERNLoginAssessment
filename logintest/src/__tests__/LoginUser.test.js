import React from "react";
import renderer from 'react-test-renderer';
import { MemoryRouter } from "react-router-dom";
import LoginUser from '../components/LoginUser.js';

it("render a login user comp", () => {

    const wrapper = renderer.create(
        <MemoryRouter>

            <LoginUser />

        </MemoryRouter>
    ).toJSON();

    expect(wrapper).toMatchSnapshot();

}); 