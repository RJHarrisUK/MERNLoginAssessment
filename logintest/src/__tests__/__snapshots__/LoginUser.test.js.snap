// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render a login user comp 1`] = `
<div>
  <h1>
    Login User
  </h1>
  <form
    onSubmit={[Function]}
  >
    <input
      id="username"
      placeholder="Username"
      type="text"
    />
    <br />
    <input
      id="email"
      placeholder="E-mail"
      type="text"
    />
    <br />
    <input
      id="password"
      placeholder="Password"
      type="text"
    />
    <p
      style={
        Object {
          "color": "red",
        }
      }
    >
      
    </p>
    <br />
    <button
      type="submit"
    >
      Login User
    </button>
  </form>
</div>
`;
